frontend/.eslintrc

```diff
@@ -43,8 +43,9 @@
           "required": {
               "some": [ "nesting", "id" ]
           }
-        }],        
-        "react/no-array-index-key": "off"
+        }],
+        "react/no-array-index-key": "off",
+        "react/function-component-definition": "off"
```

Comment:
Why add this? Feels like bad practice to have all these different ways of defining components.

Reply:
I forgot to mention it in the change log above. The rule was set to allow arrow-function component definition without invalidating the already existing function expressions. 

I wanted to propose moving to using arrow functions instead of function expressions for components since they look/feel more convenient and to write functional components, especially typed ones. E.g.:

```tsx
interface ComponentProps {
  prop1: string;
  prop2: number;
  prop3: boolean;
}

const Component: React.FC<ComponentProps> = ({ prop1, prop2, prop3 }) => {
  /* ... */
}
```

Compared to

```tsx
function Component({ prop1, prop2, prop3}: ComponentProps) {
  /* ... */
}
```

It also allows for shorter component definitions if they return only JSX:

```tsx
const Component: React.FC = () => (
  <div>
    /* ... */
  </div>
);
```

Compared to

```tsx
function Component() {
  return (
    <div>
      /* ... */
    </div>
  );
}
```

I suppose that this is also more of an opinionated decision and each developer may have their own preference on how they define components. This is why I set the rule to be `"off"` instead of specifying. At the end of the day, if we follow proper conventions for file-naming, exporting, and folder organization, there should be no confusion on reading and expecting component definitions.

Reply:
I am not very convinced that we want people to just define components any way they want. I think it makes it much harder to read if we have a bunch of different ways to define components.

I am open to which kind of definition to use but would argue we should choose only one.

Reply:
For simplicity, let us use the existing function expression then as it will require less work to move to and is not of much importance